jvm/src/main/scala/kiama/util/Server.scala

```diff
@@ -641,3 +855,12 @@ class Services[N, C <: Config, M <: Message](
 }
 
 class SetTraceNotificationParams(value: String = "off")
+
+// Notebooks
+
+case class NotebookCell(uri: String)
+
+case class Notebook(
+  uri: String,
+  cells: Map[String, NotebookCell] = Map.empty
+)
```

Comment:
Notebooks only point to the sources stored in `compiler.sources`, which are turned into `BufferSource`s in order to support incremental update.